doc/swss-schema.md

```diff
@@ -147,6 +155,213 @@ and reflects the LAG ports into the redis under: `LAG_TABLE:<team0>:port`
+	1*priority				= number
+	
+	;example1
+	; key "SCHEDULER_TABLE:SCAVENGER"
+	;	field_name					value
+	;		algorithm  				DWRR
+	;		weight					 35
+	
+	;example2
+	; key "SCHEDULER_TABLE:BEST_EFFORT"
+	;	field_name					value
+	;		algorithm		        PRIORITY
+	;		priority				7
+
+###WRED\_PROFILE\_TABLE
+	; WRED profile
+	; SAI mapping - saiwred.h
+	key						= "WRED_PROFILE_TABLE:"name
+	;fields
+	yellow_max_threshold	= byte count
+	green_max_threshold		= byte count
+	;example
+	; key == WRED_PROFILE_TABLE:AZURE
+	;	field					value
+	;	yellow_max_threshold	100
+	;	green_max_threshold		200
+
+###QUEUE\_QOS\_TABLE
+	; QOS assignments for a given port.queue
+	; SAI mapping - maps to a sai_queue object of a given port.
+	key					= "QUEUE_QOS_TABLE:" port_name ":" queue_reference
+	port_name			= name
+	queue_reference		= ref_hash_key_reference
+	
+	;field					value
+	scheduler			= 1*ref_hash_key_reference; reference to scheduler key
+	wred_profile		= 1*ref_hash_key_reference; reference to wred profile key
+
+	; example
+	; key QUEUE_QOS_TABLE:ETHERNET4:[QUEUE_TABLE:BEST_EFFORT]; setting for eth4.best_effort_queue
+	; 	field						value
+	; scheduler			[SCHEDULER_TABLE:BEST_EFFORT]
+	; wred_profile		[WRED_PROFILE_TABLE:AZURE]	
+
+###JSON configuration file
+Some network applications require network communication to build proper configuration, for example to build BGP settings.
+However there are cases where network configuration is not required, for example QOS settings. In such cases the settings will be stored in a JSON file, corresponding to the current schema and will be consumed by **swss-config**, an application which will apply the settings described in JSON file to the Redis DB (APP_DB).
+
+#### JSON sample for QOS data
+	
+	[
+		{
+			"QUEUE_TABLE:BEST_EFFORT": {
+				"queue_index": 1
+			},
+			"OP": "SET"
+		},
+		{
+			"TC_TO_QUEUE_MAP_TABLE:AZURE": {
+				"5": "[QUEUE_TABLE:BEST_EFFORT]"
+			},
+			"OP": "SET"
+		},
+		{
+			"DSCP_TO_TC_MAP_TABLE:AZURE": {
+				"7":"5",
+				"6":"5",
+				"3":"3",
+				"8":"7",			
+				"9":"8"
+			},
+			"OP": "SET"
+		},
+		{
+			"DSCP_TO_TC_MAP_TABLE:AZURE": {
+				"7":"6",			
+				"8":"7"
+			},
+			"OP": "DEL"
+		},
+		{
+			"PORT_QOS_TABLE:ETHERNET4": {
+				"dscp_to_tc_map" : "[DSCP_TO_TC_MAP_TABLE:AZURE]",
+				"tc_to_queue_map": "[TC_TO_QUEUE_MAP_TABLE:AZURE]"		
+			},
+			"OP": "SET"		
+		},
+		{
+			"SCHEDULER_TABLE:SCAVENGER" : {
+				"algorithm":"DWRR",
+				"weight": "35"
+			},
+			"OP": "SET"
+		},
+		{
+			"SCHEDULER_TABLE:BEST_EFFORT" : {
+				"algorithm":"PRIORITY",
+				"priority": "7"
+			},
+			"OP": "SET"
+		},
+		{
+			"WRED_PROFILE_TABLE:AZURE" : {
+				"yellow_max_threshold":"200",
+				"green_max_threshold": "100"
+			},
+			"OP": "SET"
+		},
+		{
+			"QUEUE_QOS_TABLE:ETHERNET4:[QUEUE_TABLE:BEST_EFFORT]" : {
+				"scheduler"		:	"[SCHEDULER_TABLE:BEST_EFFORT]",
+				"wred_profile"	: 	"[WRED_PROFILE_TABLE:AZURE]"
+			},
+			"OP": "SET"
+		}
+	]
+
+
+
+
+----------
```

Comment:
I am thinking would it be better to put move scheduler, wred_profile fields from QUEUE_QOS_TABLE into QUEUE_TABLE? It aligns with saiqueue.h definition and also address the other question raised by elad.